pkg/asset/machines/master.go

```diff
@@ -643,9 +660,61 @@ func (m *Master) Generate(ctx context.Context, dependencies asset.Parents) error
 			Data:     data,
 		}
 	}
+
+	// This is only used by Two Node OpenShift with Fencing (TNF)
+	// The credentials are rendered into secrets that will be consumed by the
+	// cluster-etcd operator to enable recovery via fencing
+	if pool.Fencing != nil && pool.Fencing.Credentials != nil {
+		credentials, err := gatherFencingCredentials(pool.Fencing.Credentials)
+		if err != nil {
+			return err
+		}
+
+		secrets, err := createSecretAssetFiles(credentials, fencingCredentialsSecretFileName)
+		if err != nil {
+			return errors.Wrap(err, "failed to gather fencing credentials for control plane hosts")
+		}
+		m.FencingCredentialsSecretFiles = append(m.FencingCredentialsSecretFiles, secrets...)
+	}
+
 	return nil
 }
 
+func gatherFencingCredentials(credentials []*types.Credential) ([]corev1.Secret, error) {
+	secrets := []corev1.Secret{}
+
+	for _, credential := range credentials {
+		if credential.HostName == "" || credential.Username == "" || credential.Password == "" || credential.Address == "" {
+			return nil, fmt.Errorf("incomplete fencing credentials for host with HostName='%s'", credential.HostName)
```

Comment:
nit/suggestion: the `%q` verb can be used instead of `%s` to have go insert the quotation marks.